File: php/testcode/benchmark_test_00530.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00530(BenchmarkRequest $req): BenchmarkResponse {
    $id = $req->param('id');
    $pdo = getDbConnection();

    $stmt = $pdo->prepare('SELECT bio FROM users WHERE id = ?');
    $stmt->execute([$id]);
    $row = $stmt->fetch();
    $bio = $row['bio'] ?? '';

    $html = '<div class="bio">' . $bio . '</div>';

    return BenchmarkResponse::html($html);
}

File: ruby/testcode/benchmark_test_00530.rb
require_relative 'shared'
require 'logger'

LOGGER = Logger.new($stdout)

def log_gsub_control_chars(req)
  sanitized = req.param('val').gsub(/[[:cntrl:]]/, '?')
  LOGGER.info(sanitized)
  BenchmarkResponse.json({ ok: true })
end
